internal/engine/actions/actions.go

```diff
@@ -138,22 +138,82 @@ func (rae *RuleActionsEngine) processAction(
 	params engif.ActionsParams,
 	metadata *json.RawMessage,
 ) (json.RawMessage, error) {
+	logger := zerolog.Ctx(ctx).Debug().
+		Str("action", string(actionType)).
+		Str("cmd", string(cmd))
+
+	// Exit early in case we shouldn't do anything, i.e., cmd != ActionCmdOn or cmd != ActionCmdOff
+	if cmd == engif.ActionCmdDoNothing {
+		logger.Msg("skipping action")
+		switch actionType {
+		case remediate.ActionType:
+			// Return the previous remediation status.
+			err := enginerr.RemediationStatusAsError(params.GetEvalStatusFromDb().RemStatus.RemediationStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().RemMetadata)), err
+		case alert.ActionType:
+			// Return the previous alert status.
+			err := enginerr.AlertStatusAsError(params.GetEvalStatusFromDb().AlertStatus.AlertStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().AlertMetadata)), err
+		}
```

Comment:
Do we need a default case?

Reply:
In particular, if there's a different action type, it drops out of the `if` and goes on to the rest of the body.

Reply:
My assumption was every action should be able to act correctly when it's called with the `ActionCmdDoNothing ` command, so even if we add a new one later and we miss to add it to this part there shouldn't be a problem. But now that you asked it it got me thinking that if we end up going to the action while skipping this part for some reason currently this will not preserve the old state.

I'll remove the `switch` here and add each case to be handled explicitly by each action. This way it's up to the action to implement what it should do on `ActionCmdDoNothing`, i.e. some may decide to preserve old state, some may choose something else.

Thanks!